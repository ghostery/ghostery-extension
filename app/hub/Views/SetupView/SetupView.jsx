/**
 * Setup View Component
 *
 * Ghostery Browser Extension
 * https://www.ghostery.com/
 *
 * Copyright 2018 Ghostery, Inc. All rights reserved.
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0
 */

import React from 'react';
import PropTypes from 'prop-types';
import { Route } from 'react-router-dom';

// Components
import SetupNavigation from '../SetupViews/SetupNavigation';
import SetupHeader from '../SetupViews/SetupHeader';

/**
 * A Functional React component for rendering the Setup View
 * @return {JSX} JSX for rendering the Setup View of the Hub app
 * @memberof HubComponents
 */
const SetupView = (props) => {
	const { extraRoutes, sendMountActions, steps } = props;

	return (
		<div className="full-height flex-container flex-dir-column">
			<div className="flex-child-grow">
				{steps.map(step => (
					<Route
						key={`route-${step.index}`}
						path={step.path}
						render={() => (
							<div>
								<SetupHeader {...step.headerProps} />
								<step.bodyComponent index={step.index} sendMountActions={sendMountActions} />
							</div>
						)}
					/>
				))}
				{extraRoutes.map(route => (
					<Route
						key={`extra-route-${route.name}`}
						path={route.path}
						render={() => <route.component sendMountActions={sendMountActions} />}
					/>
				))}
			</div>

			<SetupNavigation totalSteps={steps.length} />
		</div>
	);
};

// PropTypes ensure we pass required props of the correct type
SetupView.propTypes = {
	steps: PropTypes.arrayOf(PropTypes.shape({
		index: PropTypes.number.isRequired,
		path: PropTypes.string.isRequired,
		bodyComponent: PropTypes.func.isRequired,
		headerProps: PropTypes.shape({
			title: PropTypes.string.isRequired,
			titleImage: PropTypes.string.isRequired,
		}).isRequired,
	})).isRequired,
	extraRoutes: PropTypes.arrayOf(PropTypes.shape({
		name: PropTypes.string.isRequired,
		path: PropTypes.string.isRequired,
		component: PropTypes.func.isRequired,
	})).isRequired,
	sendMountActions: PropTypes.bool.isRequired,
};

export default SetupView;
