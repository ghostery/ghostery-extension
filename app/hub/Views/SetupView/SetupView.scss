/**
 * Setup View Sass
 *
 * Ghostery Browser Extension
 * https://www.ghostery.com/
 *
 * Copyright 2018 Ghostery, Inc. All rights reserved.
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0
 */

// Foundation Overrides
.SetupModal__buttonContainer .button {
	min-width: 160px;
	margin-bottom: 0;
	background-color: #ffffff;
}

// Setup Modal
.SetupModal__content {
	position: relative;
	width: 410px;
	min-height: 410px;
	padding-top: 45px;
	color: #4a4a4a;
	background-color: #f7f7f7;
	border: 2px solid #930194;
	z-index: 10;
}
.SetupModal__exit {
	position: absolute;
	top: -13px;
	right: -13px;
	height: 26px;
	width: 26px;
	border-radius: 50%;
	border: 1px solid #930194;
	background-color: #f7f7f7;
	background-image: buildIconX(#4a4a4a);
	background-repeat: no-repeat;
	background-position: center center;
	background-size: 12px 12px;
	cursor: pointer;
}
.SetupModal__image {
	height: 64px;
	width: 58px;
	background-size: 64px 58px;
	background-image: url('/app/images/hub/setup/ghosty-icon.svg');
	background-position: center center;
	background-repeat: no-repeat;
}
.SetupModal__text {
	font-size: 18px;
	line-height: 27px;
	font-weight: 500;
	text-align: center;
	padding: 35px 55px 15px;
}
.SetupModal__buttonContainer {
	background-color: #e7ecee;
	padding: 25px 25px 0;
}
.SetupModal__buttonContainer {
	font-size: 12px;
	color: #4a4a4a;
	cursor: pointer;
}
