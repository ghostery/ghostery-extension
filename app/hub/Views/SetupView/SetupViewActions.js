/**
 * Setup View Action creators
 *
 * Ghostery Browser Extension
 * https://www.ghostery.com/
 *
 * Copyright 2018 Ghostery, Inc. All rights reserved.
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0
 */

import { log, sendMessageInPromise } from '../../utils';
import {
	GET_SETUP_SHOW_WARNING_OVERRIDE,
	SET_SETUP_SHOW_WARNING_OVERRIDE,
	INIT_SETUP_PROPS,
	SET_SETUP_NAVIGATION
} from './SetupViewConstants';

export function getSetupShowWarningOverride() {
	return function (dispatch) {
		return sendMessageInPromise(GET_SETUP_SHOW_WARNING_OVERRIDE).then((data) => {
			dispatch({
				type: GET_SETUP_SHOW_WARNING_OVERRIDE,
				data,
			});

			// Send data back to SetupViewContainer
			return data;
		}).catch((err) => {
			log('setupView Action getSetupShowWarningOverride Error', err);
		});
	};
}

export function setSetupShowWarningOverride(actionData) {
	return function (dispatch) {
		return sendMessageInPromise(SET_SETUP_SHOW_WARNING_OVERRIDE, actionData).then((data) => {
			dispatch({
				type: SET_SETUP_SHOW_WARNING_OVERRIDE,
				data,
			});
		}).catch((err) => {
			log('setupView Action setSetupShowWarningOverride Error', err);
		});
	};
}

export function initSetupProps(data) {
	return {
		type: INIT_SETUP_PROPS,
		data,
	};
}

export function setSetupNavigation(data) {
	return {
		type: SET_SETUP_NAVIGATION,
		data,
	};
}
