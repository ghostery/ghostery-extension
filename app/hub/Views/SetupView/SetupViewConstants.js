/**
 * Custom Setup Constants
 *
 * Ghostery Browser Extension
 * https://www.ghostery.com/
 *
 * Copyright 2018 Ghostery, Inc. All rights reserved.
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0
 */

// Setup View
export const GET_SETUP_SHOW_WARNING_OVERRIDE = 'GET_SETUP_SHOW_WARNING_OVERRIDE';
export const SET_SETUP_SHOW_WARNING_OVERRIDE = 'SET_SETUP_SHOW_WARNING_OVERRIDE';
export const INIT_SETUP_PROPS = 'INIT_SETUP_PROPS';
export const SET_SETUP_NAVIGATION = 'SET_SETUP_NAVIGATION';

// Setup Blocking
export const SET_BLOCKING_POLICY = 'SET_BLOCKING_POLICY';

// Setup Anti-Suite
export const SET_ANTI_TRACKING = 'SET_ANTI_TRACKING';
export const SET_AD_BLOCK = 'SET_AD_BLOCK';
export const SET_SMART_BLOCKING = 'SET_SMART_BLOCKING';
export const SET_GHOSTERY_REWARDS = 'SET_GHOSTERY_REWARDS';

// Setup Human Web
export const SET_HUMAN_WEB = 'SET_HUMAN_WEB';
