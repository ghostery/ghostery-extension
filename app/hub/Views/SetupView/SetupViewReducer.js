/**
 * Reducer used throughout the Setup View's flow
 *
 * Ghostery Browser Extension
 * https://www.ghostery.com/
 *
 * Copyright 2018 Ghostery, Inc. All rights reserved.
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0
 */

import {
	GET_SETUP_SHOW_WARNING_OVERRIDE,
	SET_SETUP_SHOW_WARNING_OVERRIDE,
	INIT_SETUP_PROPS,
	SET_SETUP_NAVIGATION,
	SET_BLOCKING_POLICY,
	SET_ANTI_TRACKING,
	SET_AD_BLOCK,
	SET_SMART_BLOCKING,
	SET_GHOSTERY_REWARDS,
	SET_HUMAN_WEB
} from './SetupViewConstants';

const initialState = {};

function SetupViewReducer(state = initialState, action) {
	switch (action.type) {
		// Setup View
		case GET_SETUP_SHOW_WARNING_OVERRIDE: // Same as SET_SETUP_SHOW_WARNING_OVERRIDE
		case SET_SETUP_SHOW_WARNING_OVERRIDE: {
			const { setup_show_warning_override } = action.data;
			return Object.assign({}, state, {
				setup: Object.assign({}, state.setup, { setup_show_warning_override }),
			});
		}
		case INIT_SETUP_PROPS: {
			const {
				navigation,
				setup_show_warning_override,
				blockingPolicy,
				enable_anti_tracking,
				enable_ad_block,
				enable_smart_blocking,
				enable_ghostery_rewards,
				enable_human_web,
			} = action.data;
			const {
				activeIndex,
				hrefPrev,
				hrefNext,
				hrefDone,
				textPrev,
				textNext,
				textDone,
			} = navigation;
			return Object.assign({}, state, {
				setup: {
					navigation: {
						activeIndex,
						hrefPrev,
						hrefNext,
						hrefDone,
						textPrev,
						textNext,
						textDone,
					},
					setup_show_warning_override,
					blockingPolicy,
					enable_anti_tracking,
					enable_ad_block,
					enable_smart_blocking,
					enable_ghostery_rewards,
					enable_human_web,
				},
			});
		}
		case SET_SETUP_NAVIGATION: {
			const {
				activeIndex,
				hrefPrev,
				hrefNext,
				hrefDone,
				textPrev,
				textNext,
				textDone,
			} = action.data;
			return Object.assign({}, state, {
				setup: Object.assign({}, state.setup, {
					navigation: {
						activeIndex,
						hrefPrev,
						hrefNext,
						hrefDone,
						textPrev,
						textNext,
						textDone,
					},
				}),
			});
		}

		// Setup Blocking View
		case SET_BLOCKING_POLICY: {
			const { blockingPolicy } = action.data;
			return Object.assign({}, state, {
				setup: Object.assign({}, state.setup, { blockingPolicy }),
			});
		}

		// Setup Anti-Suite View
		case SET_ANTI_TRACKING: {
			const { enable_anti_tracking } = action.data;
			return Object.assign({}, state, {
				setup: Object.assign({}, state.setup, { enable_anti_tracking }),
			});
		}
		case SET_AD_BLOCK: {
			const { enable_ad_block } = action.data;
			return Object.assign({}, state, {
				setup: Object.assign({}, state.setup, { enable_ad_block }),
			});
		}
		case SET_SMART_BLOCKING: {
			const { enable_smart_blocking } = action.data;
			return Object.assign({}, state, {
				setup: Object.assign({}, state.setup, { enable_smart_blocking }),
			});
		}
		case SET_GHOSTERY_REWARDS: {
			const { enable_ghostery_rewards } = action.data;
			return Object.assign({}, state, {
				setup: Object.assign({}, state.setup, { enable_ghostery_rewards }),
			});
		}

		// Setup Human Web View
		case SET_HUMAN_WEB: {
			const { enable_human_web } = action.data;
			return Object.assign({}, state, {
				setup: Object.assign({}, state.setup, { enable_human_web }),
			});
		}

		default: return state;
	}
}

export default SetupViewReducer;
