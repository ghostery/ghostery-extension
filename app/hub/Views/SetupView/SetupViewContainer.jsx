/**
 * Setup View Container
 *
 * Ghostery Browser Extension
 * https://www.ghostery.com/
 *
 * Copyright 2019 Ghostery, Inc. All rights reserved.
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0
 */

import React, { Component } from 'react';
import { NavLink } from 'react-router-dom';
import QueryString from 'query-string';
import PropTypes from 'prop-types';
import SetupView from './SetupView';
import { Modal, ToggleCheckbox } from '../../../shared-components';
import { BLOCKING_POLICY_RECOMMENDED } from './SetupViewConstants';
import globals from '../../../../src/classes/Globals';

// Component Views
import SetupBlockingView from '../SetupViews/SetupBlockingView';
import SetupBlockingDropdown from '../SetupViews/SetupBlockingDropdown';
import SetupAntiSuiteView from '../SetupViews/SetupAntiSuiteView';
import SetupHumanWebView from '../SetupViews/SetupHumanWebView';
import SetupDoneView from '../SetupViews/SetupDoneView';

const { BROWSER_INFO } = globals;
const IS_FIREFOX = (BROWSER_INFO.name === 'firefox');

/**
 * @class Implement the Setup View for the Ghostery Hub
 * @extends Component
 * @memberof HubContainers
 */
class SetupViewContainer extends Component {
	constructor(props) {
		super(props);
		this.state = {
			sendMountActions: false,
			showModal: false,
		};

		const { history } = this.props;
		if (!props.preventRedirect) {
			history.push('/setup/1');
		}

		const title = t('hub_setup_page_title');
		window.document.title = title;

		const { actions, setup } = this.props;
		actions.setSetupStep({ setup_step: 7 });
		actions.initSetupProps(setup);
		actions.getSetupShowWarningOverride().then((data) => {
			const { setup_show_warning_override } = data;
			const { justInstalled } = QueryString.parse(window.location.search);
			const { user } = props;

			if (((justInstalled !== 'true') || user) && setup_show_warning_override) {
				this._toggleModal();
			} else {
				const { origin, pathname, hash } = window.location;
				window.history.pushState({}, '', `${origin}${pathname}${hash}`);
				this._setDefaultSettings();
			}
		});
	}

	/**
	* Function to handle clicking yes on the Modal
	*/
	_answerModalYes = () => {
		this._setDefaultSettings();
		this._toggleModal();
	}

	/**
	* Function to toggle the Modal
	*/
	_toggleModal = () => {
		const { showModal } = this.state;
		this.setState({
			showModal: !showModal,
		});
	}

	/**
	* Function to toggle the Ask Again Checkbox
	*/
	_toggleCheckbox = () => {
		const { actions, setup } = this.props;
		const { setup_show_warning_override } = setup;
		actions.setSetupShowWarningOverride({
			setup_show_warning_override: !setup_show_warning_override,
		});
	}

	/**
	 * Function to persist the default settings to background
	 */
	_setDefaultSettings() {
		this.setState({ sendMountActions: true });
		const { actions } = this.props;
		actions.setSetupStep({ setup_step: 8 });
		actions.setBlockingPolicy({ blockingPolicy: BLOCKING_POLICY_RECOMMENDED });
		actions.setAntiTracking({ enable_anti_tracking: true });
		actions.setAdBlock({ enable_ad_block: true });
		actions.setSmartBlocking({ enable_smart_block: true });
		actions.setHumanWeb({ enable_human_web: !IS_FIREFOX || BROWSER_INFO.name === 'ghostery_desktop' || BROWSER_INFO.name === 'ghostery_android' });
	}

	/**
	 * Helper render function for rendering the Modal's Children
	 * @return {JSX} JSX of the Setup Modal's Children
	 */
	_renderModalChildren() {
		const { setup } = this.props;
		const { setup_show_warning_override } = setup;

		return (
			<div className="SetupModal__content flex-container flex-dir-column align-middle">
				<div className="SetupModal__image" />
				<div className="SetupModal__text flex-child-grow">
					{t('hub_setup_enter_modal_text')}
				</div>
				<div className="SetupModal__buttonContainer full-width">
					<div className="full-width flex-container align-justify">
						<NavLink to="/home" className="button success hollow">
							{t('hub_setup_modal_button_no')}
						</NavLink>
						<div className="button success hollow" onClick={this._answerModalYes}>
							{t('hub_setup_modal_button_yes')}
						</div>
					</div>
					<div className="flex-container align-center-middle">
						<ToggleCheckbox checked={!setup_show_warning_override} onChange={this._toggleCheckbox} />
						<div className="SetupModal__checkboxText clickable" onClick={this._toggleCheckbox}>
							{t('hub_setup_modal_button_ask_again')}
						</div>
					</div>
				</div>
			</div>
		);
	}

	/**
	 * React's required render function. Returns JSX
	 * @return {JSX} JSX for rendering the Setup View of the Hub app
	 */
	render() {
		const { showModal, sendMountActions } = this.state;
		const steps = [
			{
				index: 1,
				path: '/setup/1',
				bodyComponent: SetupBlockingView,
				headerProps: {
					title: t('hub_setup_header_title_blocking'),
					titleImage: '/app/images/hub/setup/ghosty-block-all.svg',
				},
			},
			{
				index: 2,
				path: '/setup/2',
				bodyComponent: SetupAntiSuiteView,
				headerProps: {
					title: t('hub_setup_header_title_antisuite'),
					titleImage: '/app/images/hub/setup/ghosty-shield-stop-lightbulb.svg',
				},
			},
			{
				index: 3,
				path: '/setup/3',
				bodyComponent: SetupHumanWebView,
				headerProps: {
					title: t('hub_setup_header_title_humanweb'),
					titleImage: '/app/images/hub/setup/ghosty-human-web.svg',
				},
			},
			{
				index: 4,
				path: '/setup/4',
				bodyComponent: SetupDoneView,
				headerProps: {
					title: t('hub_setup_header_title_done'),
					titleImage: '/app/images/hub/setup/ghosty-check-wrench.svg',
				},
			},
		];

		const extraRoutes = [
			{
				name: '1/custom',
				path: '/setup/1/custom',
				component: SetupBlockingDropdown,
			}
		];

		return (
			<div className="full-height">
				<Modal show={showModal}>
					{this._renderModalChildren()}
				</Modal>
				<SetupView steps={steps} extraRoutes={extraRoutes} sendMountActions={sendMountActions} />
			</div>
		);
	}
}

// PropTypes ensure we pass required props of the correct type
// Note: isRequired is not needed when a prop has a default value
SetupViewContainer.propTypes = {
	preventRedirect: PropTypes.bool,
	setup: PropTypes.shape({
		navigation: PropTypes.shape({
			activeIndex: PropTypes.number,
			hrefPrev: PropTypes.oneOfType([
				PropTypes.bool,
				PropTypes.string,
			]),
			hrefNext: PropTypes.oneOfType([
				PropTypes.bool,
				PropTypes.string,
			]),
			hrefDone: PropTypes.oneOfType([
				PropTypes.bool,
				PropTypes.string,
			]),
			textPrev: PropTypes.oneOfType([
				PropTypes.bool,
				PropTypes.string,
			]),
			textNext: PropTypes.oneOfType([
				PropTypes.bool,
				PropTypes.string,
			]),
			textDone: PropTypes.oneOfType([
				PropTypes.bool,
				PropTypes.string,
			]),
		}),
		setup_show_warning_override: PropTypes.bool,
		blockingPolicy: PropTypes.string,
		enable_anti_tracking: PropTypes.bool,
		enable_ad_block: PropTypes.bool,
		enable_smart_block: PropTypes.bool,
		enable_human_web: PropTypes.bool,
	}),
	actions: PropTypes.shape({
		getSetupShowWarningOverride: PropTypes.func.isRequired,
		setSetupShowWarningOverride: PropTypes.func.isRequired,
		initSetupProps: PropTypes.func.isRequired,
		setSetupStep: PropTypes.func.isRequired,
		setSetupNavigation: PropTypes.func.isRequired,
		setBlockingPolicy: PropTypes.func.isRequired,
		setAntiTracking: PropTypes.func.isRequired,
		setAdBlock: PropTypes.func.isRequired,
		setSmartBlocking: PropTypes.func.isRequired,
		setHumanWeb: PropTypes.func.isRequired,
		setSetupComplete: PropTypes.func.isRequired,
	}).isRequired,
};

// Default props used throughout the Setup flow
SetupViewContainer.defaultProps = {
	preventRedirect: false,
	setup: {
		navigation: {
			activeIndex: 0,
			hrefPrev: false,
			hrefNext: false,
			hrefDone: false,
			textPrev: false,
			textNext: false,
			textDone: false,
		},
		setup_show_warning_override: true,
		blockingPolicy: BLOCKING_POLICY_RECOMMENDED,
		enable_anti_tracking: true,
		enable_ad_block: true,
		enable_smart_block: true,
		enable_human_web: true,
	},
};

export default SetupViewContainer;
