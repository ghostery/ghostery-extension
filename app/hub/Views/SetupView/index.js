/**
 * Point of entry index.js file for Setup View
 *
 * Ghostery Browser Extension
 * https://www.ghostery.com/
 *
 * Copyright 2019 Ghostery, Inc. All rights reserved.
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0
 */

import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { withRouter } from 'react-router-dom';

import SetupViewContainer from './SetupViewContainer';
import SetupViewReducer from './SetupViewReducer';
import * as SetupViewActions from './SetupViewActions';
import setBlockingPolicy from '../SetupViews/SetupBlockingView/SetupBlockingViewActions';
import {
	setAntiTracking,
	setAdBlock,
	setSmartBlocking
} from '../SetupViews/SetupAntiSuiteView/SetupAntiSuiteViewActions';
import setHumanWeb from '../SetupViews/SetupHumanWebView/SetupHumanWebViewActions';
import setSetupComplete from '../SetupViews/SetupDoneView/SetupDoneViewActions';

/**
 * Map redux store state properties to the component's own properties.
 * @param  {Object} state    entire Redux store's state
 * @return {function}        this function returns a plain object, which will be merged into the component's props
 * @memberof HubContainers
 */
const mapStateToProps = state => ({ ...state.setup, ...state.account });

/**
 * Bind the component's action creators using Redux's bindActionCreators.
 * @param  {function} dispatch redux store method which dispatches actions
 * @return {function}          to be used as an argument in redux connect call
 * @memberof SetupContainers
 */
const mapDispatchToProps = dispatch => ({
	actions: bindActionCreators({
		...SetupViewActions,
		setBlockingPolicy,
		setAntiTracking,
		setAdBlock,
		setSmartBlocking,
		setHumanWeb,
		setSetupComplete
	}, dispatch),
});

export const reducer = SetupViewReducer;

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(SetupViewContainer));
