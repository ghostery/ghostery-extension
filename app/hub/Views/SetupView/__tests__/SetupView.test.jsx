/**
 * Setup View Test Component
 *
 * Ghostery Browser Extension
 * https://www.ghostery.com/
 *
 * Copyright 2018 Ghostery, Inc. All rights reserved.
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0
 *
 * ToDo: Update this file with Integration tests using Enzyme
 */

import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import { MemoryRouter } from 'react-router';

// Mock Necessary Imports
jest.mock('../../SetupViews/SetupHeader', () => props => <div>Mock Setup Header</div>);
jest.mock('../../SetupViews/SetupNavigation', () => props => <div>Mock Setup Navigation</div>);

// Import Components
import SetupView from '../SetupView';

// Mock Test Component
const TestComponent = props => (
	<div>test component</div>
);

describe('app/hub/Views/SetupView component', () => {
	describe('Snapshot tests with react-test-renderer', () => {
		test('setup view is rendered correctly on first route', () => {
			const initialState = {
				steps: [
					{
						index: 1,
						path: '/test/1',
						bodyComponent: TestComponent,
						headerProps: {
							title: 'title test 1',
							titleImage: 'image for title test 1',
						}
					},
					{
						index: 2,
						path: '/test/2',
						bodyComponent: TestComponent,
						headerProps: {
							title: 'title test 2',
							titleImage: 'image for title test 2',
						}
					},
				],
			};
			const component = renderer.create(
				<MemoryRouter initialEntries={['/test/1']} >
					<SetupView {...initialState} />
				</MemoryRouter>
			).toJSON();
			expect(component).toMatchSnapshot();
		});

		test('setup view is rendered correctly on last route', () => {
			const initialState = {
				steps: [
					{
						index: 1,
						path: '/example/1',
						bodyComponent: TestComponent,
						headerProps: {
							title: 'title example 1',
							titleImage: 'image for title example 1',
						}
					},
					{
						index: 2,
						path: '/example/2',
						bodyComponent: TestComponent,
						headerProps: {
							title: 'title example 2',
							titleImage: 'image for title example 2',
						}
					},
					{
						index: 3,
						path: '/example/3',
						bodyComponent: TestComponent,
						headerProps: {
							title: 'title example 3',
							titleImage: 'image for title example 3',
						}
					},
				],
			};
			const component = renderer.create(
				<MemoryRouter initialEntries={['/example/3']} >
					<SetupView {...initialState} />
				</MemoryRouter>
			).toJSON();
			expect(component).toMatchSnapshot();
		});
	});

	describe('More Snapshot tests with react-test-renderer, but for edge cases', () => {
		test('edge case where steps is empty array', () => {
			const initialState = {
				steps: [],
			};
			const component = renderer.create(
				<MemoryRouter>
					<SetupView {...initialState} />
				</MemoryRouter>
			).toJSON();
			expect(component).toMatchSnapshot();
		});

		test('edge case where activeIndex not in steps index', () => {
			const initialState = {
				steps: [
					{
						index: 1,
						path: '/test/1',
						bodyComponent: TestComponent,
						headerProps: {
							title: 'title test 1',
							titleImage: 'image for title test 1',
						}
					},
					{
						index: 2,
						path: '/test/2',
						bodyComponent: TestComponent,
						headerProps: {
							title: 'title test 2',
							titleImage: 'image for title test 2',
						}
					},
				],
			};
			const component = renderer.create(
				<MemoryRouter initialEntries={['/test/4']} >
					<SetupView {...initialState} />
				</MemoryRouter>
			).toJSON();
			expect(component).toMatchSnapshot();
		});
	});

	describe('Mount snapshot tests rendered with Enzyme', () => {
		test.skip('the happy path of the component', () => {});

		test.skip('the non-happy path of the component', () => {});
	});
});
