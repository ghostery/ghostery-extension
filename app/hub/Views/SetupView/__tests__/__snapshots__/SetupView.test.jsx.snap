// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app/hub/Views/SetupView component More Snapshot tests with react-test-renderer, but for edge cases edge case where activeIndex not in steps index 1`] = `
<div
  className="full-height flex-container flex-dir-column"
>
  <div
    className="flex-child-grow"
  />
  <div>
    Mock Setup Navigation
  </div>
</div>
`;

exports[`app/hub/Views/SetupView component More Snapshot tests with react-test-renderer, but for edge cases edge case where steps is empty array 1`] = `
<div
  className="full-height flex-container flex-dir-column"
>
  <div
    className="flex-child-grow"
  />
  <div>
    Mock Setup Navigation
  </div>
</div>
`;

exports[`app/hub/Views/SetupView component Snapshot tests with react-test-renderer setup view is rendered correctly on first route 1`] = `
<div
  className="full-height flex-container flex-dir-column"
>
  <div
    className="flex-child-grow"
  >
    <div>
      <div>
        Mock Setup Header
      </div>
      <div>
        test component
      </div>
    </div>
  </div>
  <div>
    Mock Setup Navigation
  </div>
</div>
`;

exports[`app/hub/Views/SetupView component Snapshot tests with react-test-renderer setup view is rendered correctly on last route 1`] = `
<div
  className="full-height flex-container flex-dir-column"
>
  <div
    className="flex-child-grow"
  >
    <div>
      <div>
        Mock Setup Header
      </div>
      <div>
        test component
      </div>
    </div>
  </div>
  <div>
    Mock Setup Navigation
  </div>
</div>
`;
